From ebbf286c8f2d00878992b60970e9a3083f8709e7 Mon Sep 17 00:00:00 2001
From: TekuConcept <electro_madness@yahoo.com>
Date: Wed, 20 Oct 2021 16:04:59 -0600
Subject: [PATCH 4/4] Bug fix - remember to lock things up when closing

---
 src/node.cc | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/node.cc b/src/node.cc
index 513741b64e..b6b4d1f857 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -1179,7 +1179,16 @@ int XStep(std::shared_ptr<NodeContext> context, bool& isFinished) {
 
 void XExit(std::shared_ptr<NodeContext> context) {
   if (context == nullptr || !context->initialized) return;
-  context->main_instance->Exit(context);
+  {
+    Environment* env = context->env.get();
+    v8::Isolate* isolate = env->isolate();
+    v8::Locker locker(isolate);
+    v8::Isolate::Scope isolate_scope(isolate);
+    v8::HandleScope handle_scope(isolate);
+    v8::Context::Scope context_scope(env->context());
+    context->main_instance->Exit(context);
+    context->main_instance = nullptr;
+  }
   TearDownOncePerProcess();
   context->initialized = false;
 }
-- 
2.30.0.windows.1

